### PR TITLE
Make unit tests more pytest-onic

### PR DESCRIPTION
**Pull request by bmerry**
I wouldn't try to review everything - there are a *lot* of mechanical changes. Interesting bits are the smaller files and anything with pytest decorators (`@pytest.fixture`, `@pytest.mark`).

### CODE REVIEW

**Comment by bmerry** [comment]
The Travis failure seems to just be a transient thing where the test was too slow.

**Comment by bmerry** [comment]
Oh, and you might want to review with whitespace differences ignored, because most tests got a dedent.

File: test/test_fakeredis.py [hunk 1]
```diff
@@ -25,4529 +21,4692 @@
...
+    assert r.get('foo') == 'Ñandu'.encode()
+
+
+def test_saving_non_ascii_chars_as_key(r):
+    assert r.set('Ñandu', 'foo') is True
+    assert r.get('Ñandu') == b'foo'
+
+
+def test_saving_unicode_type_as_key(r):
+    assert r.set('Ñandu', 'foo') is True
```
Comment (ludwigschwardt) [inline test/test_fakeredis.py:168]: Should this not be `Ñandú`? 😂

File: README.rst [hunk 2]
```diff
@@ -355,43 +355,38 @@ To ensure parity with the real redis, there are a set of integration tests
 that mirror the unittests.  For every unittest that is written, the same
 test is run against a real redis instance using a real redis-py client
 instance.  In order to run these tests you must have a redis server running
-on localhost, port 6379 (the default settings).  The integration tests use
-db=10 in order to minimize collisions with an existing redis instance.
+on localhost, port 6379 (the default settings). **WARNING**: the tests will
+completely wipe your database!
```
Comment (ludwigschwardt) [inline README.rst:359]: Are you still using `db=10`? I guess a warning is in order in any case.

File: test/test_fakeredis.py [hunk 3]
```diff
@@ -25,4529 +21,4692 @@
...
+            return cls(db=db, decode_responses=decode_responses, server=fake_server)
+        else:
+            cls = getattr(redis, name)
+            return cls('localhost', port=6379, db=db, decode_responses=decode_responses)
+
+    return factory
+
+
+@pytest.fixture
+def r(request, create_redis):
```
Comment (ludwigschwardt) [inline test/test_fakeredis.py:104]: Would the `r` fixture run before each test or is it cached somehow like `self.redis` used to be?

File: test/test_fakeredis.py (hunk 1 above)
Comment (bmerry) [reply test/test_fakeredis.py:168]: No idea, it's code I inherited.

File: README.rst (hunk 2 above)
Comment (bmerry) [reply README.rst:359]: No, it hasn't used db=10 for a long time - and some tests use multiple databases.

File: test/test_fakeredis.py (hunk 3 above)
Comment (bmerry) [reply test/test_fakeredis.py:104]: It runs for each test. I don't think there was any caching before though - what caching are you referring to?

File: test/test_fakeredis.py (hunk 1 above)
Comment (ludwigschwardt) [reply test/test_fakeredis.py:168]: This is a Ñandú: https://en.wikipedia.org/wiki/Rhea_(bird)

Just wondered out loud 😄

File: test/test_fakeredis.py (hunk 3 above)
Comment (ludwigschwardt) [reply test/test_fakeredis.py:104]: Ah right, I was thinking of `self.redis` stored on the test object but that is recreated in `setup` before each test anyway. Forgot about that...

**Pull request merged**